test/e2e/helpers.go

```diff
@@ -911,3 +913,77 @@ func WaitForDaemonset(ctx context.Context, input clusterctl.ApplyClusterTemplate
 		return false
 	}, input.WaitForControlPlaneIntervals...).Should(Equal(true))
 }
+
+// podListHasNumPods fulfills the cluster-api PodListCondition type spec
+// given a list of pods, we validate for an exact number of those pods in a Running state
```

Comment:
```suggestion
// Given a list of pods, we validate for an exact number of those pods in a Running state.
```
So linter doesn't complain, and same for the other functions here.